scripts/src/java/org/oppia/android/scripts/RegexPatternValidationCheck.kt

```diff
@@ -0,0 +1,222 @@
+
+      // check if the repo has any file content failure.
+      val hasFileContentCheckFailure = retrieveFileContentChecks()
+        .fold(initial = false) { isFailing, fileContentCheck ->
+          val checkFailed = checkProhibitedContent(
+            repoPath,
+            searchFiles,
+            fileContentCheck,
+          )
+          isFailing || checkFailed
+        }
+
+      if (hasFilenameCheckFailure || hasFileContentCheckFailure) {
+        throw Exception(ScriptResultConstants.REGEX_CHECKS_FAILED)
+      } else {
+        println(ScriptResultConstants.REGEX_CHECKS_PASSED)
+      }
+    }
+    /**
+     * Retrieves all filename checks.
+     *
+     * @return a list of all the FilenameChecks
+     */
+    private fun retrieveFilenameChecks(): List<FilenameCheck> {
+      return getProto(
+        "filename_pattern_validation_checks.pb",
+        FilenameChecks.getDefaultInstance()
+      ).getFilenameChecksList()
+    }
+
+    /**
+     * Retrieves all file content checks.
+     *
+     * @return a list of all the FileContentChecks
+     */
+    private fun retrieveFileContentChecks(): List<FileContentCheck> {
+      return getProto(
+        "file_content_validation_checks.pb",
+        FileContentChecks.getDefaultInstance()
+      ).getFileContentChecksList()
+    }
+
+    /**
+     * Helper function to parse the textproto file to a proto class.
+     *
+     * @param textProtoFileName name of the textproto file to be parsed
+     * @param proto instance of the proto class
+     * @return proto class from the parsed textproto file
+     */
+    private fun <T : MessageLite> getProto(textProtoFileName: String, proto: T): T {
+      val protoBinaryFile = File("scripts/assets/$textProtoFileName")
+      val builder = proto.newBuilderForType()
+
+      @Suppress("UNCHECKED_CAST")
+      val protoObj: T =
+        FileInputStream(protoBinaryFile).use {
+          builder.mergeFrom(it)
+        }.build() as T
+      return protoObj
+    }
+
+    /**
+     * Checks for a prohibited file naming pattern.
+     *
+     * @param repoPath the path of the repo
+     * @param searchFiles a list of all the files which needs to be checked
+     * @param filenameCheck proto object of FilenameCheck
+     * @return file name pattern is correct or not
+     */
+    private fun checkProhibitedFileNamePattern(
+      repoPath: String,
+      searchFiles: List<File>,
+      filenameCheck: FilenameCheck,
+    ): Boolean {
+      val prohibitedFilenameRegex = filenameCheck.getProhibitedFilenameRegex().toRegex()
+
+      val matchedFiles = searchFiles.filter { file ->
+        return@filter file.name !in filenameCheck.getExemptedFileNameList() &&
+          prohibitedFilenameRegex.matches(
+            RepoFile.retrieveFilePath(
+              file,
+              repoPath
+            )
+          )
+      }
+
+      logProhibitedFilenameFailure(
+        repoPath,
+        filenameCheck.getFailureMessage(),
+        matchedFiles,
+      )
+
```

Comment:
no empty line needed here

Reply:
apply to other return statements as well

Reply:
Updated, PTAL.